flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/kmeans/KMeansModelParams.java

```diff
@@ -21,27 +21,11 @@
 import org.apache.flink.ml.common.param.HasDistanceMeasure;
 import org.apache.flink.ml.common.param.HasFeaturesCol;
 import org.apache.flink.ml.common.param.HasPredictionCol;
-import org.apache.flink.ml.param.IntParam;
-import org.apache.flink.ml.param.Param;
-import org.apache.flink.ml.param.ParamValidators;
 
 /**
- * Params of {@link KMeansModel}.
+ * Params of {@link KMeansModel} and {@link OnlineKMeansModel}.
  *
  * @param <T> The class type of this instance.
  */
 public interface KMeansModelParams<T>
-        extends HasDistanceMeasure<T>, HasFeaturesCol<T>, HasPredictionCol<T> {
-
-    Param<Integer> K =
```

Comment:
Would it still be useful for users to query the k of the KMeansModel (and clustering algorithm in general)?

Note that Spark's `python/pyspark/mlib/clustering.py` line 54 shows example python code snippet that prints the k.

Reply:
In `KMeansModel`, `getK()` is never used, so if the input model data's K is different from that of KMeansModel's K parameter, it would still function correctly. Because of this I removed K from `KMeansModelParams`.

Now I agree that users may want to get to know the K of the model operator. But in order to solve the concern above, I'd also like to check the model data centroids length and the value of K parameter in `KMeansModel` and `OnlineKMeansModel`, so as to make sure the K functions correctly.

Reply:
Sounds good. Let's do the check in both `KMeansModel` and `OnlineKMeansModel`.